### PR TITLE
TSK-1975 new indices for Object_Reference Table

### PR DESCRIPTION
**Pull request by arolfes**
3 new Indices for Table Object_Reference
* IDX_OBJECT_REFERENCE_PK_ID
* IDX_OBJECT_REFERENCE_FK_TASK_ID
* IDX_OBJECT_REFERENCE_ACCESS_LIST

Fixed Default Sorting when searching or accessing ObjectReferences

Sonar Cloud Results: https://sonarcloud.io/summary/new_code?id=arolfes_taskana&branch=TSK-1975-index-object-reference-table

Must be reviewed and integrated after [TSK-1971 support for oracle databases](https://github.com/Taskana/taskana/pull/2050)

---
Release Notes:

```
TSK-1975 new indices for Object_Reference Table
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: common/taskana-common/src/main/resources/sql/db2/taskana-schema-db2.sql [hunk 1]
```diff
@@ -430,3 +430,14 @@ CREATE INDEX IDX_WORKBASKET_ACCESS_LIST_WORKBASKET_ID ON WORKBASKET_ACCESS_LIST
...
 COMMIT WORK;
+
+CREATE INDEX IDX_OBJECT_REFERENCE_PK_ID ON OBJECT_REFERENCE
+    (ID ASC) ALLOW REVERSE SCANS COLLECT SAMPLED DETAILED STATISTICS;
+COMMIT WORK ;
+CREATE INDEX IDX_OBJECT_REFERENCE_FK_TASK_ID ON OBJECT_REFERENCE
+    (TASK_ID ASC) ALLOW REVERSE SCANS COLLECT SAMPLED DETAILED STATISTICS;
+COMMIT WORK ;
+CREATE INDEX IDX_OBJECT_REFERENCE_ACCESS_LIST ON OBJECT_REFERENCE
+    (COMPANY ASC, SYSTEM ASC, SYSTEM_INSTANCE ASC, TYPE ASC, VALUE ASC)
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/resources/sql/db2/taskana-schema-db2.sql:441]: "VALUE" ASC, "TYPE" ASC, "SYSTEM_INSTANCE" ASC, "SYSTEM" ASC, "COMPANY" ASC, "ID" ASC

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/ObjectReferenceMapper.java [hunk 2]
```diff
@@ -20,6 +20,7 @@ public interface ObjectReferenceMapper {
       "<script>SELECT ID, COMPANY, SYSTEM, SYSTEM_INSTANCE, TYPE, VALUE "
           + "FROM OBJECT_REFERENCE "
           + "WHERE ID = #{id}"
+          + "ORDER BY COMPANY ASC, SYSTEM ASC, SYSTEM_INSTANCE ASC, TYPE ASC, VALUE ASC"
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/ObjectReferenceMapper.java:23]: This does not match with the new order in the indeces

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/ObjectReferenceMapper.java (hunk 2 above)
Comment (arolfes) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/ObjectReferenceMapper.java:23]: i forgot this...


Done

File: common/taskana-common/src/main/resources/sql/db2/taskana_schema_update_5.2.0_to_5.10.0_db2.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+insert into TASKANA_SCHEMA_VERSION (VERSION, CREATED) values ('5.10.0', current_timestamp);
+
+create index IDX_OBJECT_REFERENCE_PK_ID on OBJECT_REFERENCE
+    (ID asc) ALLOW REVERSE SCANS COLLECT SAMPLED DETAILED STATISTICS;
+commit work ;
+create index IDX_OBJECT_REFERENCE_FK_TASK_ID on OBJECT_REFERENCE
+    (TASK_ID asc) ALLOW REVERSE SCANS COLLECT SAMPLED DETAILED STATISTICS;
+commit work ;
+create index IDX_OBJECT_REFERENCE_ACCESS_LIST on OBJECT_REFERENCE
+    (value asc, TYPE asc, SYSTEM_INSTANCE asc, SYSTEM asc, COMPANY asc, ID asc)
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/resources/sql/db2/taskana_schema_update_5.2.0_to_5.10.0_db2.sql:13]:
```suggestion
    (VALUE asc, TYPE asc, SYSTEM_INSTANCE asc, SYSTEM asc, COMPANY asc, ID asc)
```

File: common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_5.2.0_to_5.10.0_h2.sql [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+insert into TASKANA_SCHEMA_VERSION (VERSION, CREATED) values ('5.10.0', current_timestamp);
+
+create index IDX_OBJECT_REFERENCE_PK_ID on OBJECT_REFERENCE
+    (ID asc);
+commit work ;
+create index IDX_OBJECT_REFERENCE_FK_TASK_ID on OBJECT_REFERENCE
+    (TASK_ID asc);
+commit work ;
+create index IDX_OBJECT_REFERENCE_ACCESS_LIST on OBJECT_REFERENCE
+    (value asc, TYPE asc, SYSTEM_INSTANCE asc, SYSTEM asc, COMPANY asc, ID asc);
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/resources/sql/h2/taskana_schema_update_5.2.0_to_5.10.0_h2.sql:13]:
```suggestion
    (VALUE asc, TYPE asc, SYSTEM_INSTANCE asc, SYSTEM asc, COMPANY asc, ID asc);
```

File: common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_5.2.0_to_5.10.0_postgres.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+insert into TASKANA_SCHEMA_VERSION (VERSION, CREATED) values ('5.10.0', current_timestamp);
+
+create index IDX_OBJECT_REFERENCE_PK_ID on OBJECT_REFERENCE
+    (ID asc);
+commit work ;
+create index IDX_OBJECT_REFERENCE_FK_TASK_ID on OBJECT_REFERENCE
+    (TASK_ID asc);
+commit work ;
+create index IDX_OBJECT_REFERENCE_ACCESS_LIST on OBJECT_REFERENCE
+    (value asc, TYPE asc, SYSTEM_INSTANCE asc, SYSTEM asc, COMPANY asc, ID asc);
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/resources/sql/postgres/taskana_schema_update_5.2.0_to_5.10.0_postgres.sql:14]:
```suggestion
    (VALUE asc, TYPE asc, SYSTEM_INSTANCE asc, SYSTEM asc, COMPANY asc, ID asc);
```

**Review by mustaphazorgati — APPROVED** [review]
Comments are not necessary, but would be nice :) Can be intermixed with further PRs and don't have to be resolved now.

**Comment by arolfes** [comment]
Must be reviewed and integrated after https://github.com/Taskana/taskana/pull/2050

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**